server/routes/api/series.py

```diff
@@ -80,34 +76,63 @@ def get_binned_series(entities, variables, year):
   
   Returns:
     JSON response from server, with series containing only observations from 
-    the specified year.
+    the specified year, binned by month. The format of the JSON looks like:
+
+    {
+      'data': [
+        {
+          "date": "YYYY-MM",
+          "value": 1234
+        }, ...
+      ],
+      "facet": {
+        "importName": "Name",
+          "provenanceUrl": "https://provenance.url/",
+          "measurementMethod": "MeasurementMethod",
+          "unit": "unit"
+      }
+    }
   """
-  # Get raw series from mixer
+  # Get raw api response from mixer
   data = series_core(entities, variables, False)
 
   for stat_var in variables:
     for location in data['data'][stat_var].keys():
 
-      # filter series to just observations from the selected year
-      series = data['data'][stat_var][location]['series']
-      pruned_series = []
-      dates_with_data = []
+      series = data.get('data', {}).get(stat_var, {}).get(location,
+                                                          {}).get('series', [])
+      data_found = False
+
+      # Get the latest month we have data for
+      latest_date = series[-1].get('date', '')
+      latest_month = 12
+      if latest_date[:4] == year and len(latest_date) >= 7:
+        latest_month = int(latest_date[5:7])
```

Comment:
nit: try to avoid "magic numbers", even if it's for something like date parsing.